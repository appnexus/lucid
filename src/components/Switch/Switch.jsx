import _ from 'lodash';
import React from 'react';
import PropTypes from 'prop-types';
import { lucidClassNames } from '../../util/style-helpers';
import {
	createClass as createReactClass,
	omitProps,
} from '../../util/component-types';

const cx = lucidClassNames.bind('&-Switch');
const { bool, func, object, string } = PropTypes;

/**
 * {"categories": ["controls", "toggles"]}
 *
 * This is a toggle -- a component that is in one of two particular states at
 * any given moment in time -- that uses a visualization of a physical on/off
 * switch made popular by smartphone OSes to reflect its current state.
 *
 * It uses a hidden native check box control under the hood but leverages other
 * HTML elements to visualize its state.
 */
const Switch = createReactClass({
	displayName: 'Switch',
	propTypes: {
		/**
		 * Appended to the component-specific class names set on the root
		 * element.
		 */
		className: string,

		/**
		 * Indicates whether the component should appear and act disabled by
		 * having a "greyed out" palette and ignoring user interactions.
		 */
		isDisabled: bool,

		/**
		 * Indicates that the component is in the "selected" state when true
		 * and in the "unselected" state when false.
		 */
		isSelected: bool,

		/**
		 * Called when the user clicks on the component or when they press the
		 * space key while the component is in focus.
		 *
		 * Signature: `(isSelected, { event, props }) => {}`
		 */
		onSelect: func,

		/**
		 * Passed through to the root element.
		 */
		style: object,
	},

	getDefaultProps() {
		return {
			isDisabled: false,
			isSelected: false,
			onSelect: _.noop,
		};
	},

	componentDidMount() {
		this.nativeElement = this.refs.nativeElement;
	},

	render() {
		const {
			className,
			isDisabled,
			isSelected,
			style,
			...passThroughs
		} = this.props;

		return (
			<span
				className={cx(
					'&',
					{
						'&-is-disabled': isDisabled,
						'&-is-selected': isSelected,
					},
					className
				)}
				onClick={this.handleClicked}
				onTouchEnd={this.handleClicked}
				style={style}
			>
				<input
					onChange={_.noop}
					{...omitProps(passThroughs, Switch, 'children')}
					checked={isSelected}
					className={cx('&-native')}
					disabled={isDisabled}
					ref="nativeElement"
					type="checkbox"
				/>
				<span className={cx('&-visualization-container')} />
				<span className={cx('&-visualization-glow')} />
				<span className={cx('&-visualization-handle')} />
			</span>
		);
	},

	handleClicked(event) {
		const { isDisabled, isSelected, onSelect } = this.props;

		event.preventDefault();

		if (!isDisabled) {
			onSelect(!isSelected, { event, props: this.props });
			this.nativeElement.focus();
		}
	},
});

export default Switch;
