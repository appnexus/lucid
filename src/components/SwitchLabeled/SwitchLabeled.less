@import (reference) '../../styles/variables.less';
@import (reference) '../../styles/mixins.less';

.lucid-SwitchLabeled {
	align-items: center;
	color: @color-darkGray;
	cursor: pointer;
	display: flex;
	font: @font;
	justify-content: flex-start;

	> .lucid-Switch {
		flex: 0 0 @Switch-width;
		margin-right: @size-XXS;
	}

	&&-is-disabled {
		color: @color-mediumGray;
		cursor: not-allowed;
	}

	&:hover .lucid-Switch-visualization-glow {
		.opacity();
		transform: @Switch-glow-scale;
	}

	&:hover .lucid-Switch-visualization-handle {
		transform: @Switch-handle-scale;
	}

	&&-is-disabled:hover .lucid-Switch-visualization-glow {
		.opacity(0);
		transform: none;
	}

	&&-is-disabled:hover .lucid-Switch-visualization-handle {
		transform: none;
	}

	&-is-disabled&-is-selected:hover .lucid-Switch-visualization-handle {
		transform: @Switch-selected-transform scale(1);
	}


	&-is-selected:hover .lucid-Switch-visualization-glow {
		transform: @Switch-selected-transform @Switch-glow-scale;
	}

	&-is-selected:hover .lucid-Switch-visualization-handle {
		transform: @Switch-selected-transform @Switch-handle-scale;
	}

	&-text {
		.transition-group-animation-fade(@timing-animation-hover);
		position: relative;
		height: @Switch-height;

		// Since we have two elements in the dom during the transition from one
		// label to another, we want the element that's leaving to appear "over
		// the top" of the label that coming in.
		&-leave {
			position: absolute;
			top: 0;
			left: 0;
		}

		> * {
			white-space: nowrap;
			line-height: @Switch-height;
		}
	}
}
