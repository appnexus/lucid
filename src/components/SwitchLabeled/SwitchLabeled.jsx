import _ from 'lodash';
import React from 'react';
import PropTypes from 'prop-types';
import ReactTransitionGroup from 'react-transition-group/CSSTransitionGroup';
import { lucidClassNames } from '../../util/style-helpers';
import {
	createClass as createReactClass,
	getFirst,
	omitProps,
} from '../../util/component-types';
import Switch from '../Switch/Switch';

const cx = lucidClassNames.bind('&-SwitchLabeled');
const { any, node, object, string } = PropTypes;

/**
 * {"categories": ["controls", "toggles"], "extend": "Switch", "madeFrom": ["Switch"]}
 *
 * This is a composite of the `Switch` component and the native `label`
 * element.
 */
const SwitchLabeled = createReactClass({
	displayName: 'SwitchLabeled',

	components: {
		/**
		 * Label to be shown alongside the switch.
		 */
		Label: createReactClass({
			displayName: 'SwitchLabeled.Label',
			propName: 'Label',
			propTypes: {
				/**
				 * Used to identify the purpose of this switch to the user -- can be
				 * any renderable content.
				 */
				children: node,
			},
		}),
	},

	propTypes: {
		...Switch.propTypes,

		/**
		 * Appended to the component-specific class names set on the root
		 * element.
		 */
		className: string,

		/**
		 * Passed through to the root element.
		 */
		style: object,

		/**
		 * Child element whose children are used to identify the purpose of this
		 * switch to the user.
		 */
		Label: any,
	},

	getDefaultProps() {
		return {
			isDisabled: false,
			isSelected: false,
			onSelect: _.noop,
		};
	},

	componentWillMount() {
		this._labelKey = 0;
	},

	componentWillReceiveProps(nextProps) {
		const currentLabel = _.get(
			getFirst(this.props, SwitchLabeled.Label),
			'props.children',
			null
		);
		const nextLabel = _.get(
			getFirst(nextProps, SwitchLabeled.Label),
			'props.children',
			null
		);

		if (currentLabel !== nextLabel) {
			this._labelKey++;
		}
	},

	render() {
		const {
			className,
			isDisabled,
			isSelected,
			onSelect,
			style,
			...passThroughs
		} = this.props;

		const labelChildProps = _.get(
			getFirst(this.props, SwitchLabeled.Label),
			'props'
		);

		return (
			<label
				className={cx(
					'&',
					{
						'&-is-disabled': isDisabled,
						'&-is-selected': isSelected,
					},
					className
				)}
				style={style}
			>
				<Switch
					className={className}
					isDisabled={isDisabled}
					isSelected={isSelected}
					onSelect={onSelect}
					{...omitProps(passThroughs, SwitchLabeled, [], false)}
				/>
				<ReactTransitionGroup
					transitionName={cx('&-text')}
					transitionEnterTimeout={100}
					transitionLeaveTimeout={100}
					style={{ position: 'relative' }}
					className={cx('&-text')}
				>
					{labelChildProps
						? <span key={this._labelKey}>
								{labelChildProps.children || labelChildProps}
							</span>
						: null}
				</ReactTransitionGroup>
			</label>
		);
	},
});

export default SwitchLabeled;
