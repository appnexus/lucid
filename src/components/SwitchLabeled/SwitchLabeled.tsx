import _, { omit } from 'lodash';
import React from 'react';
import PropTypes from 'prop-types';
import { CSSTransition } from 'react-transition-group';

import { lucidClassNames } from '../../util/style-helpers';
import { getFirst, StandardProps } from '../../util/component-types';
import Switch from '../Switch/Switch';
import { useState } from 'react';
import { useEffect } from 'react';

const cx = lucidClassNames.bind('&-SwitchLabeled');
const { any, node, object, string } = PropTypes;

const defaultProps = {
	isDisabled: false,
	isSelected: false,
	onSelect: _.noop,
};

const defaultState = {
	_labelKey: 0,
};

export interface ISwitchLabeledProps extends StandardProps {
	isDisabled: boolean;
	isSelected: boolean;
	onSelect: any;
}

const SwitchLabeled = (props: ISwitchLabeledProps) => {
	const {
		className,
		isDisabled,
		isSelected,
		onSelect,
		style,
		...passThroughs
	} = props;

	const [state, setState] = useState(defaultState);

	const currentLabel = _.get(
		getFirst(props, SwitchLabeled.Label),
		'props.children',
		null
	);

	useEffect(() => {
		setState({ _labelKey: state._labelKey + 1 });
	}, [currentLabel]);

	const labelChildProps = _.get(getFirst(props, SwitchLabeled.Label), 'props');

	const isShown = !!labelChildProps;

	return (
		<label
			className={cx(
				'&',
				{
					'&-is-disabled': isDisabled,
					'&-is-selected': isSelected,
				},
				className
			)}
			style={style}
		>
			<Switch
				className={className}
				isDisabled={isDisabled}
				isSelected={isSelected}
				onSelect={onSelect}
				{...omit(
					passThroughs,
					[
						'className',
						'style',
						'Label',
						'isDisabled',
						'isSelected',
						'onSelect',
						'isIncludeExclude',
					].concat('initialState')
				)}
			/>
			{labelChildProps && (
				<CSSTransition
					in={isShown}
					classNames={cx('&-text')}
					timeout={100}
					style={{ position: 'relative' }}
					className={cx('&-text')}
					unmountOnExit
				>
					<span key={state._labelKey}>
						{labelChildProps.children || labelChildProps}
					</span>
				</CSSTransition>
			)}
		</label>
	);
};

SwitchLabeled.defaultProps = defaultProps;

SwitchLabeled.displayName = 'SwitchLabeled';

SwitchLabeled.peek = {
	description: `A composite of the \`Switch\` component and the native \`label\` element.`,
	categories: ['controls', 'toggles'],
	madeFrom: ['Switch'],
};

const Label = () => {
	return null;
};

Label.displayName = 'SwitchLabeled.Label';

Label.peek = {
	description: `Label to be shown alongside the \`Switch\`.`,
};

Label.propName = 'Label';

Label.propTypes = {
	/**
		Used to identify the purpose of this switch to the user -- can be any
		renderable content.
	*/
	children: node,
};

SwitchLabeled.propTypes = {
	...Switch.propTypes,

	/**
		Appended to the component-specific class names set on the root element.
	*/
	className: string,

	/**
		Passed through to the root element.
	*/
	style: object,

	/**
		Child element whose children are used to identify the purpose of this
		switch to the user.
	*/
	Label: any,
};

SwitchLabeled.Label = Label;

export default SwitchLabeled;
