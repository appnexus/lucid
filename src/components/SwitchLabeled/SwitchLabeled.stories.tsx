import React, { useState } from 'react';
import { Meta, Story } from '@storybook/react';

import SwitchLabeled, { ISwitchLabeledProps } from './SwitchLabeled';

export default {
	title: 'Controls/SwitchLabeled',
	component: SwitchLabeled,
	subcomponents: { 'SwitchLabeled.Label': SwitchLabeled.Label },
	parameters: {
		docs: {
			description: {
				component: (SwitchLabeled as any).peek.description,
			},
		},
	},
	args: SwitchLabeled.defaultProps,
} as Meta;

export const Basic: Story<ISwitchLabeledProps> = (args) => {
	const [isSelected, setIsSelected] = useState(false);

	const handleSelected = () => {
		setIsSelected(!isSelected);
	};

	return (
		<SwitchLabeled
			{...args}
			Label='Default'
			onSelect={handleSelected}
			isSelected={isSelected}
		/>
	);
};

/* Interactive */
export const Interactive: Story<ISwitchLabeledProps> = (args) => {
	const style = {
		marginBottom: '3px',
	};

	const [state, setState] = useState({
		airplaneMode: false,
		bluetooth: false,
		cellularData: false,
	});

	const handleSelectedAirplaneMode = (isSelected: any) => {
		setState({
			...state,
			airplaneMode: isSelected,
		});
	};

	const handleSelectedBluetooth = (isSelected: any) => {
		setState({
			...state,
			bluetooth: isSelected,
		});
	};

	const handleSelectedCellularData = (isSelected: any) => {
		setState({
			...state,
			cellularData: isSelected,
		});
	};

	return (
		<section>
			<p>
				<em>
					(Use the styles on the parent container of <code>SwitchLabeled</code>{' '}
					components to ensure only the switches and their labels are clickable)
				</em>
			</p>
			<span
				style={{
					display: 'inline-flex',
					flexDirection: 'column',
					alignItems: 'flex-start',
				}}
			>
				<SwitchLabeled
					{...args}
					isSelected={state.airplaneMode === true}
					onSelect={handleSelectedAirplaneMode}
					style={style}
				>
					<SwitchLabeled.Label>Airplane Mode</SwitchLabeled.Label>
				</SwitchLabeled>
				<SwitchLabeled
					{...args}
					isSelected={state.bluetooth === true}
					onSelect={handleSelectedBluetooth}
					style={style}
				>
					<SwitchLabeled.Label>Bluetooth</SwitchLabeled.Label>
				</SwitchLabeled>
				<SwitchLabeled
					{...args}
					isSelected={state.cellularData === true}
					onSelect={handleSelectedCellularData}
					style={style}
				>
					<SwitchLabeled.Label>Cellular Data</SwitchLabeled.Label>
				</SwitchLabeled>
			</span>
		</section>
	);
};

/* Interactive With Changing Labels */
export const InteractiveWithChangingLabels: Story<ISwitchLabeledProps> = (
	args
) => {
	const style = {
		marginBottom: '3px',
	};

	const [state, setState] = useState({
		airplaneMode: false,
		bluetooth: false,
		cellularData: false,
		spam: false,
	});

	const handleSelectedAirplaneMode = (isSelected: any) => {
		setState({
			...state,
			airplaneMode: isSelected,
		});
	};

	const handleSelectedBluetooth = (isSelected: any) => {
		setState({
			...state,
			bluetooth: isSelected,
		});
	};

	const handleSelectedCellularData = (isSelected: any) => {
		setState({
			...state,
			cellularData: isSelected,
		});
	};

	const handleSelectedSpam = (isSelected: any) => {
		setState({
			...state,
			spam: isSelected,
		});
	};

	const spamSwitchLabel = state.spam
		? 'Yes! I would like to receive updates, special offers, and other information from Xandr and its subsidiaries.'
		: 'No! Please keep your messages to yourself!';

	return (
		<section>
			<span
				style={{
					display: 'inline-flex',
					flexDirection: 'column',
					alignItems: 'flex-start',
				}}
			>
				<SwitchLabeled
					{...args}
					isSelected={state.airplaneMode === true}
					Label={`Airplane Mode ${
						state.airplaneMode === true ? 'Activated' : 'Deactivated'
					}`}
					onSelect={handleSelectedAirplaneMode}
					style={style}
				/>
				<SwitchLabeled
					{...args}
					isSelected={state.bluetooth === true}
					Label={`Bluetooth ${
						state.bluetooth === true ? 'Enabled' : 'Disabled'
					}`}
					onSelect={handleSelectedBluetooth}
					style={style}
				/>
				<SwitchLabeled
					{...args}
					isSelected={state.cellularData === true}
					Label={`${state.cellularData ? 'Use' : 'Do Not Use'} Cellular Data`}
					onSelect={handleSelectedCellularData}
					style={style}
				/>
			</span>
			<br />
			<SwitchLabeled
				{...args}
				isSelected={state.spam === true}
				onSelect={handleSelectedSpam}
				style={style}
			>
				<SwitchLabeled.Label>{spamSwitchLabel}</SwitchLabeled.Label>
			</SwitchLabeled>
		</section>
	);
};

/* Label As Prop */
export const LabelAsProp: Story<ISwitchLabeledProps> = (args) => {
	const style = {
		marginRight: '5px',
	};

	return (
		<section>
			<section
				style={{
					display: 'inline-flex',
					flexDirection: 'column',
					alignItems: 'flex-start',
				}}
			>
				<SwitchLabeled {...args} Label='Just text' style={style} />
				<SwitchLabeled
					{...args}
					Label={<span>HTML element</span>}
					style={style}
				/>
				<SwitchLabeled
					{...args}
					Label={[
						'Text in an array',
						'Only the first value in the array is used',
						'The rest of these should be ignored',
					]}
					style={style}
				/>
				<SwitchLabeled
					{...args}
					Label={[
						<span key='1'>HTML element in an array</span>,
						<span key='2'>
							Again only the first value in the array is used
						</span>,
						<span key='3'>The rest should not be rendered</span>,
					]}
					style={style}
				/>
			</section>
		</section>
	);
};

/* States */
export const States: Story<ISwitchLabeledProps> = (args) => {
	const style = {
		marginBottom: '3px',
		marginRight: '13px',
	};

	return (
		<section
			style={{
				display: 'inline-flex',
				flexDirection: 'column',
				alignItems: 'flex-start',
			}}
		>
			<SwitchLabeled {...args} style={style}>
				<SwitchLabeled.Label>(default props)</SwitchLabeled.Label>
			</SwitchLabeled>

			<section>
				<SwitchLabeled {...args} isSelected={true} style={style}>
					<SwitchLabeled.Label>Selected</SwitchLabeled.Label>
				</SwitchLabeled>
				<SwitchLabeled {...args} isDisabled={true} style={style}>
					<SwitchLabeled.Label>Disabled</SwitchLabeled.Label>
				</SwitchLabeled>
				<SwitchLabeled
					{...args}
					isDisabled={true}
					isSelected={true}
					style={style}
				>
					<SwitchLabeled.Label>Disabled & selected</SwitchLabeled.Label>
				</SwitchLabeled>
			</section>
		</section>
	);
};
