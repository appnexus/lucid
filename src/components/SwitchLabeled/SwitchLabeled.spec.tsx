import _ from 'lodash';
import assert from 'assert';
import React from 'react';
import { CSSTransition } from 'react-transition-group';
import { mount, shallow } from 'enzyme';

import { common } from '../../util/generic-tests';

import SwitchLabeled from './SwitchLabeled';
import Switch from '../Switch/Switch';

describe('SwitchLabeled', () => {
	common(SwitchLabeled);

	describe('props', () => {
		describe('isDisabled', () => {
			it('passes the value through to its `Switch` instance.', () => {
				const wrapper = shallow(<SwitchLabeled isDisabled={true} />);

				assert.equal(wrapper.find(Switch).prop('isDisabled'), true);
			});

			it('defaults to `false`.', () => {
				const wrapper = mount(<SwitchLabeled />);

				assert.equal(wrapper.prop('isDisabled'), false);
			});
		});

		describe('isSelected', () => {
			it('passes the value through to its `Switch` instance.', () => {
				const wrapper = shallow(<SwitchLabeled isSelected={true} />);

				assert.equal(wrapper.find(Switch).prop('isSelected'), true);
			});

			it('defaults to `false`.', () => {
				const wrapper = mount(<SwitchLabeled />);

				assert.equal(wrapper.prop('isSelected'), false);
			});
		});

		describe('onSelect', () => {
			it('passes the value through to its `Switch` instance.', () => {
				const foo = () => null;
				const wrapper = shallow(<SwitchLabeled onSelect={foo} />);

				assert.equal(wrapper.find(Switch).prop('onSelect'), foo);
			});

			it('defaults to the Lodash `noop` method.', () => {
				const wrapper = mount(<SwitchLabeled />);

				assert.equal(wrapper.prop('onSelect'), _.noop);
			});
		});

		describe('Label (as a prop)', () => {
			it('renders the value in a `span` neighboring its `Switch` instance.', () => {
				const wrapper = shallow(<SwitchLabeled Label={'foo'} />);

				assert.equal(
					wrapper.find(CSSTransition).find('span').prop('children'),
					'foo'
				);
			});
		});

		describe('Label (as a child)', () => {
			it('renders the value in a `span` neighboring its `Switch` instance.', () => {
				const wrapper = shallow(
					<SwitchLabeled>
						<SwitchLabeled.Label>foo</SwitchLabeled.Label>
					</SwitchLabeled>
				);

				assert.equal(
					wrapper.find(CSSTransition).find('span').prop('children'),
					'foo'
				);
			});
		});

		describe('pass throughs', () => {
			it('passes through all props not defined in `propTypes` to its `Switch` instance.', () => {
				const extraProps = {
					foo: 1,
					bar: 2,
					baz: 3,
					qux: 4,
					quux: 5,
				};

				const wrapper = shallow(
					<SwitchLabeled
						className='wut'
						isDisabled={true}
						isSelected={true}
						style={{ fontWeight: 'bold' }}
						onSelect={_.noop}
						{...extraProps}
					/>
				);
				const switchProps = wrapper.find(Switch).props();

				// It should pass `foo`, `bar`, `baz`, `qux`, and `quux` through
				// to the `Switch` instance.
				_.forEach(['foo', 'bar', 'baz', 'qux', 'quux'], (prop) => {
					assert(_.has(switchProps, prop));
				});
			});
		});
	});
});
