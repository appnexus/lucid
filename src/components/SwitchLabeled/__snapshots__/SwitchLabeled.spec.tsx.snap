// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchLabeled [common] example testing should match snapshot(s) for interactive-default 1`] = `
<label
  className="lucid-SwitchLabeled"
  style={
    Object {
      "marginBottom": "3px",
    }
  }
>
  <Switch
    isDisabled={false}
    isIncludeExclude={false}
    isSelected={false}
    onSelect={[Function]}
  >
    <SwitchLabeled.Label>
      Airplane Mode
    </SwitchLabeled.Label>
  </Switch>
  <CSSTransition
    className="lucid-SwitchLabeled-text"
    classNames="lucid-SwitchLabeled-text"
    in={true}
    style={
      Object {
        "position": "relative",
      }
    }
    timeout={100}
    unmountOnExit={true}
  >
    <span
      key="0"
    >
      Airplane Mode
    </span>
  </CSSTransition>
</label>
`;

exports[`SwitchLabeled [common] example testing should match snapshot(s) for interactive-default 2`] = `
<label
  className="lucid-SwitchLabeled"
  style={
    Object {
      "marginBottom": "3px",
    }
  }
>
  <Switch
    isDisabled={false}
    isIncludeExclude={false}
    isSelected={false}
    onSelect={[Function]}
  >
    <SwitchLabeled.Label>
      Bluetooth
    </SwitchLabeled.Label>
  </Switch>
  <CSSTransition
    className="lucid-SwitchLabeled-text"
    classNames="lucid-SwitchLabeled-text"
    in={true}
    style={
      Object {
        "position": "relative",
      }
    }
    timeout={100}
    unmountOnExit={true}
  >
    <span
      key="0"
    >
      Bluetooth
    </span>
  </CSSTransition>
</label>
`;

exports[`SwitchLabeled [common] example testing should match snapshot(s) for interactive-default 3`] = `
<label
  className="lucid-SwitchLabeled"
  style={
    Object {
      "marginBottom": "3px",
    }
  }
>
  <Switch
    isDisabled={false}
    isIncludeExclude={false}
    isSelected={false}
    onSelect={[Function]}
  >
    <SwitchLabeled.Label>
      Cellular Data
    </SwitchLabeled.Label>
  </Switch>
  <CSSTransition
    className="lucid-SwitchLabeled-text"
    classNames="lucid-SwitchLabeled-text"
    in={true}
    style={
      Object {
        "position": "relative",
      }
    }
    timeout={100}
    unmountOnExit={true}
  >
    <span
      key="0"
    >
      Cellular Data
    </span>
  </CSSTransition>
</label>
`;

exports[`SwitchLabeled [common] example testing should match snapshot(s) for interactive-with-changing-labels 1`] = `
<label
  className="lucid-SwitchLabeled lucid-SwitchLabeled-is-selected"
  style={
    Object {
      "marginBottom": "3px",
    }
  }
>
  <Switch
    isDisabled={false}
    isIncludeExclude={false}
    isSelected={true}
    onSelect={[Function]}
  />
  <CSSTransition
    className="lucid-SwitchLabeled-text"
    classNames="lucid-SwitchLabeled-text"
    in={true}
    style={
      Object {
        "position": "relative",
      }
    }
    timeout={100}
    unmountOnExit={true}
  >
    <span
      key="0"
    >
      Airplane Mode Activated
    </span>
  </CSSTransition>
</label>
`;

exports[`SwitchLabeled [common] example testing should match snapshot(s) for interactive-with-changing-labels 2`] = `
<label
  className="lucid-SwitchLabeled lucid-SwitchLabeled-is-selected"
  style={
    Object {
      "marginBottom": "3px",
    }
  }
>
  <Switch
    isDisabled={false}
    isIncludeExclude={false}
    isSelected={true}
    onSelect={[Function]}
  />
  <CSSTransition
    className="lucid-SwitchLabeled-text"
    classNames="lucid-SwitchLabeled-text"
    in={true}
    style={
      Object {
        "position": "relative",
      }
    }
    timeout={100}
    unmountOnExit={true}
  >
    <span
      key="0"
    >
      Bluetooth Enabled
    </span>
  </CSSTransition>
</label>
`;

exports[`SwitchLabeled [common] example testing should match snapshot(s) for interactive-with-changing-labels 3`] = `
<label
  className="lucid-SwitchLabeled lucid-SwitchLabeled-is-selected"
  style={
    Object {
      "marginBottom": "3px",
    }
  }
>
  <Switch
    isDisabled={false}
    isIncludeExclude={false}
    isSelected={true}
    onSelect={[Function]}
  />
  <CSSTransition
    className="lucid-SwitchLabeled-text"
    classNames="lucid-SwitchLabeled-text"
    in={true}
    style={
      Object {
        "position": "relative",
      }
    }
    timeout={100}
    unmountOnExit={true}
  >
    <span
      key="0"
    >
      Use Cellular Data
    </span>
  </CSSTransition>
</label>
`;

exports[`SwitchLabeled [common] example testing should match snapshot(s) for interactive-with-changing-labels 4`] = `
<label
  className="lucid-SwitchLabeled lucid-SwitchLabeled-is-selected"
  style={
    Object {
      "marginBottom": "3px",
    }
  }
>
  <Switch
    isDisabled={false}
    isIncludeExclude={false}
    isSelected={true}
    onSelect={[Function]}
  >
    <SwitchLabeled.Label>
      Yes! I would like to receive updates, special offers, and other information from AppNexus and its subsidiaries.
    </SwitchLabeled.Label>
  </Switch>
  <CSSTransition
    className="lucid-SwitchLabeled-text"
    classNames="lucid-SwitchLabeled-text"
    in={true}
    style={
      Object {
        "position": "relative",
      }
    }
    timeout={100}
    unmountOnExit={true}
  >
    <span
      key="0"
    >
      Yes! I would like to receive updates, special offers, and other information from AppNexus and its subsidiaries.
    </span>
  </CSSTransition>
</label>
`;

exports[`SwitchLabeled [common] example testing should match snapshot(s) for label-as-prop 1`] = `
<label
  className="lucid-SwitchLabeled"
  style={
    Object {
      "marginRight": "5px",
    }
  }
>
  <Switch
    isDisabled={false}
    isIncludeExclude={false}
    isSelected={false}
    onSelect={[Function]}
  />
  <CSSTransition
    className="lucid-SwitchLabeled-text"
    classNames="lucid-SwitchLabeled-text"
    in={true}
    style={
      Object {
        "position": "relative",
      }
    }
    timeout={100}
    unmountOnExit={true}
  >
    <span
      key="0"
    >
      Just text
    </span>
  </CSSTransition>
</label>
`;

exports[`SwitchLabeled [common] example testing should match snapshot(s) for label-as-prop 2`] = `
<label
  className="lucid-SwitchLabeled"
  style={
    Object {
      "marginRight": "5px",
    }
  }
>
  <Switch
    isDisabled={false}
    isIncludeExclude={false}
    isSelected={false}
    onSelect={[Function]}
  />
  <CSSTransition
    className="lucid-SwitchLabeled-text"
    classNames="lucid-SwitchLabeled-text"
    in={true}
    style={
      Object {
        "position": "relative",
      }
    }
    timeout={100}
    unmountOnExit={true}
  >
    <span
      key="0"
    >
      <span>
        HTML element
      </span>
    </span>
  </CSSTransition>
</label>
`;

exports[`SwitchLabeled [common] example testing should match snapshot(s) for label-as-prop 3`] = `
<label
  className="lucid-SwitchLabeled"
  style={
    Object {
      "marginRight": "5px",
    }
  }
>
  <Switch
    isDisabled={false}
    isIncludeExclude={false}
    isSelected={false}
    onSelect={[Function]}
  />
  <CSSTransition
    className="lucid-SwitchLabeled-text"
    classNames="lucid-SwitchLabeled-text"
    in={true}
    style={
      Object {
        "position": "relative",
      }
    }
    timeout={100}
    unmountOnExit={true}
  >
    <span
      key="0"
    >
      Text in an array
    </span>
  </CSSTransition>
</label>
`;

exports[`SwitchLabeled [common] example testing should match snapshot(s) for label-as-prop 4`] = `
<label
  className="lucid-SwitchLabeled"
  style={
    Object {
      "marginRight": "5px",
    }
  }
>
  <Switch
    isDisabled={false}
    isIncludeExclude={false}
    isSelected={false}
    onSelect={[Function]}
  />
  <CSSTransition
    className="lucid-SwitchLabeled-text"
    classNames="lucid-SwitchLabeled-text"
    in={true}
    style={
      Object {
        "position": "relative",
      }
    }
    timeout={100}
    unmountOnExit={true}
  >
    <span
      key="0"
    >
      <span
        key="1"
      >
        HTML element in an array
      </span>
    </span>
  </CSSTransition>
</label>
`;

exports[`SwitchLabeled [common] example testing should match snapshot(s) for states 1`] = `
<label
  className="lucid-SwitchLabeled"
  style={
    Object {
      "marginBottom": "3px",
      "marginRight": "13px",
    }
  }
>
  <Switch
    isDisabled={false}
    isIncludeExclude={false}
    isSelected={false}
    onSelect={[Function]}
  >
    <SwitchLabeled.Label>
      (default props)
    </SwitchLabeled.Label>
  </Switch>
  <CSSTransition
    className="lucid-SwitchLabeled-text"
    classNames="lucid-SwitchLabeled-text"
    in={true}
    style={
      Object {
        "position": "relative",
      }
    }
    timeout={100}
    unmountOnExit={true}
  >
    <span
      key="0"
    >
      (default props)
    </span>
  </CSSTransition>
</label>
`;

exports[`SwitchLabeled [common] example testing should match snapshot(s) for states 2`] = `
<label
  className="lucid-SwitchLabeled lucid-SwitchLabeled-is-selected"
  style={
    Object {
      "marginBottom": "3px",
      "marginRight": "13px",
    }
  }
>
  <Switch
    isDisabled={false}
    isIncludeExclude={false}
    isSelected={true}
    onSelect={[Function]}
  >
    <SwitchLabeled.Label>
      Selected
    </SwitchLabeled.Label>
  </Switch>
  <CSSTransition
    className="lucid-SwitchLabeled-text"
    classNames="lucid-SwitchLabeled-text"
    in={true}
    style={
      Object {
        "position": "relative",
      }
    }
    timeout={100}
    unmountOnExit={true}
  >
    <span
      key="0"
    >
      Selected
    </span>
  </CSSTransition>
</label>
`;

exports[`SwitchLabeled [common] example testing should match snapshot(s) for states 3`] = `
<label
  className="lucid-SwitchLabeled lucid-SwitchLabeled-is-disabled"
  style={
    Object {
      "marginBottom": "3px",
      "marginRight": "13px",
    }
  }
>
  <Switch
    isDisabled={true}
    isIncludeExclude={false}
    isSelected={false}
    onSelect={[Function]}
  >
    <SwitchLabeled.Label>
      Disabled
    </SwitchLabeled.Label>
  </Switch>
  <CSSTransition
    className="lucid-SwitchLabeled-text"
    classNames="lucid-SwitchLabeled-text"
    in={true}
    style={
      Object {
        "position": "relative",
      }
    }
    timeout={100}
    unmountOnExit={true}
  >
    <span
      key="0"
    >
      Disabled
    </span>
  </CSSTransition>
</label>
`;

exports[`SwitchLabeled [common] example testing should match snapshot(s) for states 4`] = `
<label
  className="lucid-SwitchLabeled lucid-SwitchLabeled-is-disabled lucid-SwitchLabeled-is-selected"
  style={
    Object {
      "marginBottom": "3px",
      "marginRight": "13px",
    }
  }
>
  <Switch
    isDisabled={true}
    isIncludeExclude={false}
    isSelected={true}
    onSelect={[Function]}
  >
    <SwitchLabeled.Label>
      Disabled & selected
    </SwitchLabeled.Label>
  </Switch>
  <CSSTransition
    className="lucid-SwitchLabeled-text"
    classNames="lucid-SwitchLabeled-text"
    in={true}
    style={
      Object {
        "position": "relative",
      }
    }
    timeout={100}
    unmountOnExit={true}
  >
    <span
      key="0"
    >
      Disabled & selected
    </span>
  </CSSTransition>
</label>
`;
